Lib/idlelib/idle_test/test_sidebar.py

```diff
@@ -370,5 +378,164 @@ def assert_colors_are_equal(colors):
         assert_colors_are_equal(orig_colors)
 
 
+class TestShellSidebar(unittest.TestCase):
+    root: tk.Tk = None
+    shell: PyShell = None
+
+    @classmethod
+    def setUpClass(cls):
+        requires('gui')
+
+        idlelib.pyshell.use_subprocess = True
+
+        cls.root = root = tk.Tk()
+
```

Comment:
```suggestion
        root.withdraw()
```
Tests still pass without extra window.

Reply:
Good catch!